lib/SILOptimizer/Transforms/ArrayElementValuePropagation.cpp

```diff
@@ -206,10 +269,16 @@ bool ArrayAllocation::recursivelyCollectUses(ValueBase *Def) {
 
     // Check array semantic calls.
     ArraySemanticsCall ArrayOp(User);
-    if (ArrayOp && ArrayOp.doesNotChangeArray()) {
-      if (ArrayOp.getKind() == ArrayCallKind::kGetElement)
+    if (ArrayOp) {
```

Comment:
Could this be refactored out into a function returning a boolean or something like that?